prow/plugins/commands/generics/generics.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package generics
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"k8s.io/test-infra/prow/errorutil"
+	"k8s.io/test-infra/prow/pluginhelp"
+	"k8s.io/test-infra/prow/plugins"
+	"k8s.io/test-infra/prow/plugins/commands"
+	"k8s.io/test-infra/prow/plugins/commands/handlers"
+	"k8s.io/test-infra/prow/plugins/commands/policies"
+)
+
+const reFormatAddRemove = `(?mi)^/(remove-)?%s(?: +(.*?))?\s*$`
+
+// AddRemoveCommands take the form `/[remove-]<command> [<args>]`.
+func AddRemoveCommand(command string, issueType commands.IssueType, add, remove commands.MatchHandler) *commands.Command {
+	handler := func(ctx *commands.Context) error {
+		var addMatches, removeMatches [][]string
+		for _, match := range ctx.Matches {
+			if len(match) != 3 {
```

Comment:
Why are we concerned this won't be three?

Reply:
It should always be 3 unless there is a bug, but if there is a bug an error is better than a segfault.
I can remove this if desired though, I don't feel strongly about it and it really shouldn't have a length other than 3 unless the regular expression is messed up.